src/components/datalist-item/DatalistItem.js

```diff
@@ -53,7 +54,7 @@ class DatalistItem extends React.Component<Props> {
         // required aria props are added dynamically
         /* eslint-disable jsx-a11y/role-has-required-aria-props */
         return (
-            <li {...itemProps} className={classes} id={this.id} role="option">
+            <li {...itemProps} className={classes} id={this.id} role="option" title={rest.title}>
```

Comment:
rest is already part of itemProps, it will automatically apply the title, don't need to specify is explicitly